hedera-mirror-rest/transactions.js

```diff
@@ -766,17 +773,51 @@ const extractSqlFromTransactionsByIdOrHashRequest = async (transactionIdOrHash,
   }
 
   mainConditions.unshift(...commonConditions);
-  return {query: getTransactionQuery(mainConditions.join(' and '), commonConditions.join(' and ')), params};
+  return {
+    query: getTransactionQuery(mainConditions.join(' and '), commonConditions.join(' and ')),
+    params,
+    scheduledParamExists: scheduledParamExists,
+    isTransactionHash: isTransactionHash,
+  };
 };
 
+function getTransactionsByIdOrHashCacheControlHeader(transactionsRows, isTransactionHash, scheduled) {
+  if (scheduled || isTransactionHash) {
```

Comment:
As I mentioned [previously](https://github.com/hashgraph/hedera-mirror-node/pull/9327#discussion_r1760252662).
```suggestion
function getTransactionsByIdOrHashCacheControlHeader(transactionsRows) {
  if (transactionsRows.length < 2) {
```

Reply:
not entirely true. it's possible to have `schedule=false` query param and return multiple transactions for the same transaction id. Then we can have a successful schedulecreate in the response with all others failed, ideally we should use the longer max-age because the existence of the `schedule=?` query param.

However given that such scenario happens rarely, and using a shorter max-age doesn't hurt function, agree we can go with the simpler approach.

Reply:
I missed that this scheduled variable represents the query param exists and not that it exists and is set to true, hence my comment. We can keep the scheduled check but omit isTransactionHash and just use list size for that.